src/prefect/tasks/sendgrid/sendgrid.py

```diff
@@ -0,0 +1,140 @@
+from pathlib import Path
+from typing import Any, List, Union
+
+from prefect.client import Secret
+from prefect.core import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+class SendEmail(Task):
+    """
+    A task for sending an email via Twilio SendGrid. For this task to
+    function properly, you must have a Prefect Secret set which stores
+    your SendGrid API Key (defaults to `"SENDGRID_API_KEY"`).
+
+    Args:
+        - from_email (str): The email address of the sender; defaults to notifications@prefect.io
+        - to_emails (str, tuple, list(str), list(tuple)): The email address of the recipient(s);
+            can also be provided at runtime.
+            Refer to [SendGrid-Python](https://github.com/sendgrid/sendgrid-python) for specifics.
+        - subject (str, optional): The subject of the email; can also be provided at runtime
+        - html_content (str): The html body of the email; can also be provided at runtime
+        - category (Union[str, List[str]], optional): The category/categories to use for the email;
+            can also be provided at runtime
+        - attachment_file_path (Union[str, Path], optional): The file path of the email attachment;
+            can also be provided at runtime
+        - sendgrid_secret (str, optional): the name of the Prefect Secret which stores your
+            SendGrid API key; defaults to `"SENDGRID_API_KEY"`
+        - **kwargs (optional): additional kwargs to pass to the `Task` constructor
+    """
+
+    def __init__(
+        self,
+        from_email: str = "notifications@prefect.io",
+        to_emails: str = None,
```

Comment:
Can we update this to reflect all available types? `Union[str, Tuple, List[str], List[Tuple]]`